crowbar_framework/app/helpers/barclamp/neutron_helper.rb

```diff
@@ -58,16 +38,28 @@ def networking_ml2_type_drivers_for_neutron(selected)
       valid_options.each do |el|
         options << [el, el]
       end
-      options_for_select(options, selected)
+      options
+    end
+
+    def networking_ml2_mechanism_drivers_for_neutron(selected)
+      selected = selected.gsub(/\s+/, "").split(",")
+      valid_options = NeutronService.networking_ml2_mechanism_drivers_valid
+      options_for_select(NeutronHelper.options_keep_order(valid_options, selected), selected)
+    end
+
+    def networking_ml2_type_drivers_for_neutron(selected)
+      selected = selected.gsub(/\s+/, "").split(",")
+      valid_options = NeutronService.networking_ml2_type_drivers_valid
+      options_for_select(NeutronHelper.options_keep_order(valid_options, selected), selected)
     end
 
     def networking_ml2_type_drivers_default_provider_network_for_neutron(selected)
-      options_for_select(networking_ml2_type_drivers_valid().map{|x| [x, x]},
+      options_for_select(NeutronService.networking_ml2_type_drivers_valid().map{|x| [x, x]},
                          selected.to_s)
     end
 
     def networking_ml2_type_drivers_default_tenant_network_for_neutron(selected)
-      options_for_select(networking_ml2_type_drivers_valid().map{|x| [x, x]},
+      options_for_select(NeutronService.networking_ml2_type_drivers_valid().map{|x| [x, x]},
```

Comment:
Done in https://github.com/crowbar/barclamp-neutron/commit/ed72a9a35db400e62101fa774ed32947ee5880b1